pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -373,10 +373,10 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
         this.topicLoadRequestSemaphore = new AtomicReference<Semaphore>(
                 new Semaphore(pulsar.getConfiguration().getMaxConcurrentTopicLoadRequest(), false));
         if (pulsar.getConfiguration().getMaxUnackedMessagesPerBroker() > 0
-                && pulsar.getConfiguration().getMaxUnackedMessagesPerSubscriptionOnBrokerBlocked() > 0.0) {
+                && pulsar.getConfiguration().getMaxUnackedMessagesPerSubscriptionOnBrokerBlocked() > 0) {
             this.maxUnackedMessages = pulsar.getConfiguration().getMaxUnackedMessagesPerBroker();
-            this.maxUnackedMsgsPerDispatcher = (int) ((maxUnackedMessages
-                    * pulsar.getConfiguration().getMaxUnackedMessagesPerSubscriptionOnBrokerBlocked()) / 100);
+            this.maxUnackedMsgsPerDispatcher = maxUnackedMessages
+                    * pulsar.getConfiguration().getMaxUnackedMessagesPerSubscriptionOnBrokerBlocked() / 100;
```

Comment:
This bug seems to have existed too long; maybe the user doesn't config `maxUnackedMessagesPerBroker.`
The fix is good to me. But could we simplify by just not dividing 100？

Reply:
These two ways make sense.

As I  describe in the motivation, both the Javadoc and the UT take this configuration as a percentage which is type int.

> But could we simplify by just not dividing 100？
We also need to change the UT parameters for  DispatcherBlockConsumerTest like
```java
final double unAckMsgPercentagePerDispatcher = 10;
```
to
```java
final double unAckMsgPercentagePerDispatcher = 0.1;
```